packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -67,12 +67,15 @@ export function state(v) {
 /**
  * @template V
  * @param {V} initial_value
+ * @param {boolean} [immutable]
  * @returns {Source<V>}
  */
 /*#__NO_SIDE_EFFECTS__*/
-export function mutable_source(initial_value) {
+export function mutable_source(initial_value, immutable = false) {
```

Comment:
Rather than doing this, can't we just use the non-mutable source instead?

Reply:
No, because this method additionally adds the source to the before/afterUpdate array, which is necessary in legacy mode